### PR TITLE
Feature/use string for compilation options

### LINKED ISSUE #13106: Update compilation option for signature kind
check for "mainnet"  and "testnet"  strings instead of boolean [here](https://github.com/MinaProtocol/mina/blob/8d2c5869aed7a19d90bedc5b41c944c1d790ca78/src/lib/signature_kind/compile_config/mina_signature_kind.ml)  and update the [mlh files](https://github.com/MinaProtocol/mina/tree/8d2c5869aed7a19d90bedc5b41c944c1d790ca78/src/config/features) accordingly

### PR DESCRIPTION
**Pull request by ghost-not-in-the-shell**
Explain your changes:
This PR updates the compilation options to use string for the network.

Explain how you tested your changes:
*

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #13106

### CODE REVIEW

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by bkase — APPROVED** [review]

**Comment by bkase** [comment]
!approved-for-mainnet

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/signature_kind/compile_config/mina_signature_kind.ml [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 type t = Testnet | Mainnet
 
-[%%if mainnet]
+[%%if network = "mainnet"]
 
 let t = Mainnet
 
-[%%else]
+[%%elif network = "testnet"]
```
Comment (psteckler) [inline src/lib/signature_kind/compile_config/mina_signature_kind.ml:9]: Would it be useful to have an `else` case that generates a compile-time error, if `network` is neither "mainnet" nor "testnet"? That would catch a bug in the config file.

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

File: src/lib/signature_kind/compile_config/mina_signature_kind.ml (hunk 1 above)
Comment (ghost-not-in-the-shell) [reply src/lib/signature_kind/compile_config/mina_signature_kind.ml:9]: Matthew was asking me to add another variant `Other_of_string` to catch all the other cases and I pushed a commit to do that.

**Review by psteckler — APPROVED** [review]
I would change `Other_of_string` to `Other_network`, but otherwise LGTM

**Comment by ghost-not-in-the-shell** [comment]
!ci-build-me

**Review by joaosreis — APPROVED** [review]

**Review by fabrizio-m — APPROVED** [review]

**Review by mrmr1993 — CHANGES_REQUESTED** [review]
This seems different from the design that we discussed, in that it adds a single new network type 'other' instead of the additional domain separations that we wanted. Can you tweak `Other_network` to carry the string from the config, and pass that around -- to the hash prefixes etc. -- as appropriate?

**Pull request closed without merging**